### PR TITLE
DAOS-14157 test: Update storage timeouts based upon config

### PR DESCRIPTION
**Pull request by phender**
Functional test timeouts for the storage prepare and format steps in bringing up the server engines are now be based upon the number of NVMe devices defined in the server configuration.

Skip-unit-tests: true
Skip-fault-injection-test: true
Allow-unstable-test: true
Test-tag: pr HarnessSetupTest

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by phender** [comment]
Updated code is working as expected on VM clusters:
```
2023-08-17 21:36:32,979 server_utils_bas L0153 DEBUG| Updating pattern timeout based upon server config
2023-08-17 21:36:32,980 server_utils_bas L0157 DEBUG|   Detected bdev_list entries: []
2023-08-17 21:36:32,980 command_utils_ba L0196 DEBUG| Updated param DaosServerCommand.pattern_timeout => 40
```
and HW clusters:
```
2023-08-18 00:13:15,568 server_utils_bas L0153 DEBUG| Updating pattern timeout based upon server config
2023-08-18 00:13:15,568 server_utils_bas L0157 DEBUG|   Detected bdev_list entries: ['0000:85:05.5', '0000:d7:05.5']
2023-08-18 00:13:15,569 command_utils_ba L0196 DEBUG| Updated param DaosServerCommand.pattern_timeout => 40
```

**Comment by phender** [comment]
The single failure in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12847/6/testReport/ is due to:
```
server ERR  src/engine/init.c:288 dss_tgt_nr_check() cannot start engine with 8 targets 4 xs_helpers on 6 cores, may try with DAOS_TARGET_OVERSUBSCRIBE=1 or reduce #targets/#nr_xs_helpers(2 cores reserved for system service).
```

**Comment by phender** [comment]
Running on `wolf-[248-250,321]`:
```
2023-08-18 15:25:39,191 server_utils_bas L0153 DEBUG| Updating pattern timeout based upon server config
2023-08-18 15:25:39,192 server_utils_bas L0157 DEBUG|   Detected bdev_list entries: ['0000:2b:00.0', '0000:2c:00.0', '0000:2d:00.0', '0000:2e:00.0', '0000:63:00.0', '0000:64:00.0', '0000:65:00.0', '0000:66:00.0', '0000:a2:00.0', '0000:a3:00.0', '0000:a4:00.0', '0000:a5:00.0', '0000:de:00.0', '0000:df:00.0', '0000:e0:00.0', '0000:e1:00.0']
2023-08-18 15:25:39,192 command_utils_ba L0196 DEBUG| Updated param DaosServerCommand.pattern_timeout => 320
```

File: src/tests/ftest/util/server_utils_base.py [hunk 1]
```diff
@@ -146,6 +147,20 @@ def update_pattern(self, mode, host_qty):
             self.pattern = self.NORMAL_PATTERN
         self.pattern_count = host_qty * len(self.yaml.engine_params)
 
+    def update_pattern_timeout(self):
```
Comment (daltonbohning) [inline src/tests/ftest/util/server_utils_base.py:150]: Super nit - 1. "update" commonly implies that this function will accept some argument from the caller to update something. 2. it also only updates if not already set. I wonder if something like `update_default_pattern_timeout`, `set_default_pattern_timeout`, etc. might be more clear.

**Review by mjean308 — APPROVED** [review]

**Review by daltonbohning — DISMISSED** [review]

**Comment by phender** [comment]
The only test failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12847/10/testReport/ are known issue https://daosio.atlassian.net/browse/DAOS-14208

**Review by daltonbohning — APPROVED** [review]

**Review by mjean308 — APPROVED** [review]

**Pull request merged**